wordpress-mobile/WordPress-iOS
# Reader: update to use org id to determine P2 sites

Ref: https://github.com/wordpress-mobile/WordPress-iOS/issues/15343, https://github.com/wordpress-mobile/WordPress-iOS/issues/15264
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/313

This updates `ReaderPost` and `ReaderSiteTopic` to use a site's `organization_id` to determine if it is a P2 or not.

This resolves two issues:
- [P2 stream header](https://github.com/wordpress-mobile/WordPress-iOS/pull/15442): a Followed P2 will now show the P2 header.
- [Post detail](https://github.com/wordpress-mobile/WordPress-iOS/pull/15447#issuecomment-740269725): a Followed P2 will now show the P2 default icon.

To test:

---
P2 stream:
- Log in with an account that is only following a P2.
- Go to Reader > P2.
- Verify the post header is the [P2 format](https://github.com/wordpress-mobile/WordPress-iOS/pull/15315) (i.e. displays the author information).

<kbd><img width="435" alt="p2_card_header" src="https://user-images.githubusercontent.com/1816888/101693020-49714a00-3a2e-11eb-9939-1b9b5c9fce36.png"></kbd>

---
P2 post detail:
- Log in with an account that is only following a P2 that does not have a site icon set.
- Go to Reader > P2.
- Tap on the P2 card header.
- Verify the site icon is the P2 default.

<kbd><img width="436" alt="p2_default_icon" src="https://user-images.githubusercontent.com/1816888/101693403-cac8dc80-3a2e-11eb-83e2-9a5aec8fff69.png"></kbd>

---
PR submission checklist:
